assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -1669,58 +1715,98 @@ private String build_student_view_submission_context(VelocityPortlet portlet, Co
             List<Reference> currentAttachments = (List<Reference>) state.getAttribute(ATTACHMENTS);
 
             if (s != null) {
-                log.debug("BUILD SUBMISSION FORM HAS SUBMISSION FOR USER {}", submitter);
-                context.put("submission", s);
-                String currentUser = userDirectoryService.getCurrentUser().getId();
-                String grade = assignmentService.getGradeForSubmitter(s, currentUser);
-                context.put("grade", grade);
+                // figure out if attachments have been modified
+                // the attachments from the previous submission
+                Set<String> submittedAttachments = s.getAttachments();
+                newAttachments = areAttachmentsModified(submittedAttachments, currentAttachments);
+            }else {
+                // new submission
+                // if assignment is a group submission... send group id and not user id
+                String submitterId;
                 if (assignment.getIsGroup()) {
-                    context.put("selectedGroup", s.getGroupId());
-                    context.put("originalGroup", s.getGroupId());
-                    context.put("submitterId", s.getGroupId());
+                    submitterId = assignmentService.getSubmitterIdForAssignment(assignment, user);
+                } else {
+                    submitterId = submitter.getId();
+                }
+                try {
+                    s = assignmentService.addSubmission(assignment.getId(), submitterId);
+                    if (s != null ) {
+                        setResubmissionProperties(assignment, s);
+                        s.setSubmitted(true);
+                        s.setUserSubmission(false);
+                        s.setDateModified(Instant.now());
+                        s.getSubmitters().stream().filter(sb -> sb.getSubmitter().equals(submitterId)).findFirst().ifPresent(sb -> sb.setSubmittee(false));
```

Comment:
findAny over findFirst

Reply:
findFirst() has been changed to findAny() in every place where it appeared.